Suggested upgrade to the SerpApiClient.get_html() method

gutoarraes opened the pull request:
My suggestion to fix the [open issue 66](https://github.com/serpapi/google-search-results-python/issues/66).

Since the returned object from `requests.get` isn't necessarily an HTML, I approached this by accessing the JSON returned from `get_dictionary` and performing another `requests.get` on the `raw_html_file`.

Also added an assertion in `test_google_search.py` to confirm there is a `<html>` tag present in the returned data.

gutoarraes commented:
![app](https://github.com/serpapi/google-search-results-python/assets/55930275/fba8c5a9-7878-4f50-8356-c904ccac9d0e)

* Current `get_html` method returns:

![before](https://github.com/serpapi/google-search-results-python/assets/55930275/df53a061-1f3f-4aa4-bc91-f65f34552660)

* Result with suggested changes:

![after](https://github.com/serpapi/google-search-results-python/assets/55930275/fdf1ac0c-b534-47d5-b054-cc1eead7b78d)

I'm not sure if the `<html>`  tag is a definitive requirement for all HTML pages. Therefore, the assertion I added to `tests/test_google_search.py` may not be appropriate. Appreciate if anyone can confirm/deny.

hartator commented:
@gutoarraes Sorry about that!

You should be able to grad directly the HTML from just the ID. No need to download the JSON first. The URL should be always structured like this: https://serpapi.com/searches/5b50d58a304bda2fca30bac9.json?api_key=f0776d63edb8dd607154f126eb2c07049ef4bc06caf6f621d5065be2458a607d.

Do you want to update your PR in this way or should we do it? Thank you!

gutoarraes commented:
Hey @hartator, thanks for clarifying! If you wouldn't mind I'd like to give it a crack.

hartator commented:
Cool, thank you!

Meant https://serpapi.com/searches/5b50d58a304bda2fca30bac9.html?api_key=f0776d63edb8dd607154f126eb2c07049ef4bc06caf6f621d5065be2458a607d instead of https://serpapi.com/searches/5b50d58a304bda2fca30bac9.json?api_key=f0776d63edb8dd607154f126eb2c07049ef4bc06caf6f621d5065be2458a607d in my earlier comment of course.

ilyazub commented on serpapi/serp_api_client.py:
```diff
@@ -61,11 +61,11 @@ def get_results(self, path='/search'):
 
-    def get_html(self):
+    def get_html(self, path='/search.html'):
         """Returns:
             Raw HTML search result from Google
         """
-        return self.get_results()
+        return self.get_response(path).text
```
Let's use the same approach as in the `get_json` method.

```suggestion
    def get_html(self):
        """Returns:
            Raw HTML search result from Google
        """
        self.params_dict["output"] = "html"
        return self.get_results()
```

ilyazub commented on tests/test_google_search.py:
```diff
@@ -60,6 +60,7 @@ def test_get_html(self):
 				search = GoogleSearch({"q": "Coffee", "location": "Austin,Texas"})
 				data = search.get_html()
 				self.assertGreater(len(data), 10)
+				self.assertIn("<html>", data)
```
`<html>` tag can have attributes. Let's fix this test.

![image](https://github.com/serpapi/google-search-results-python/assets/282605/2181d041-a84b-4164-b483-4dde4d6c3625)


```suggestion
				self.assertIn("<html", data)
```

gutoarraes commented:
I was cracking my head a bit about this just to realize I had already changed the PR but didn't update the description here. I removed the accessing of the JSON and simply appendended `.html` to the path which yields the same result. 

Let me know if there is a better approach, but I believe this may be what you're talking about.

ilyazub commented on tests/test_google_search.py:
```diff
@@ -60,6 +60,7 @@ def test_get_html(self):
 				data = search.get_html()
 				self.assertGreater(len(data), 10)
+				self.assertIn("<html>", data)
 
```
Let's double-check that `get_dict` still returns dictionary after calling `get_html`.

```suggestion

				data = search.get_dict()
				self.assertEqual(data["search_metadata"]["status"], "Success")
				self.assertIsNone(data.get("error"))
```

ilyazub reviewed (commented):
@gutoarraes Thank you for your help!

gutoarraes reviewed (commented):
Thank you @ilyazub for the suggestions. I've applied everything and it's now ready for review. 

Will start looking for the my next contribution. Cheers!

ilyazub reviewed (approved):
Works and looks good. Thank you, @gutoarraes.
